deobfuscator/comm.js

```diff
@@ -140,25 +156,80 @@ function concatStrings(ast) {
                 delete parentNode.left;
                 delete parentNode.right;
                 delete parentNode.operator;
-                
-                /* We can't just set nodes like node = { type: "Literal", value: val, raw: "\"" + val + "\"" };
-                  It seems that, the node passed to the enter method does not have a reference to the original AST, but accessing its properties are done by referencing the original AST.		
-                */
             }
         }
     });
 
-    return ast;	
-    //ToDo: currently it does not concatenate a + "b" + "c".
+    return ast;
 }
 
 function beautify(code){
     return escodegen.generate(concatStrings(esprima.parse(code)));
 }
 
+/**
+ * Calls eval in a sandboxed iframe.
+ * Usage: var seval = new SandboxEval(window.location.href);
+ * seval.getResult(code);
+ *  
+ * @constructor
+ */
+
+var SandboxEval = function () {
+    // The below function is a setup to run in an iframe. Receives messages, evaluates it, and pass back the result.
+    function receiveAndPassBack(){
+        function evalScript(evt){
+            //Check that the message came from where we expect.
+            if(evt.origin != "%ORIGIN%") {
+                return;
+            }
+            try{
+                parent.postMessage({"success": 1, "result":_eval(evt.data)}, "%ORIGIN%");
```

Comment:
When posting result, you'd better reuse the `requestId` from the suggested message format.